test/uts/controller_test/controller_test.go

```diff
@@ -0,0 +1,361 @@
+	{"pod_exc", "missing_objmeta_uid", true, testPoolConf1, nil},
+	{"pod_exc", "bad_deviceID_format", false, testPoolConf1, []string{"E"}},
+	{"pod_exc", "no_cpuset_file", true, testPoolConf1, nil},
+	{"pod_exc", "no_CID", true, testPoolConf1, nil},
+	{"pod_exc", "naming_mismatch", false, testPoolConf1, []string{"E"}},
+}
+
+func TestCreateController(t *testing.T) {
+	testSh := setupTestSethandler(testPoolConf1)
+	ctrl := testSh.CreateController()
+	if ctrl == nil {
+		t.Errorf("Controller creation failed")
+	}
+}
+
+func TestNew(t *testing.T) {
+	sh, err := sethandler.New(testKubeconfPath, testPoolConf1, getFakeCpusetRoot(wildcardFakeCpuSetPath))
+	if err != nil || sh == nil {
+		t.Errorf("Sethandler object creation failed because: %s", err.Error())
+	}
+}
+
+func TestPodAdded(t *testing.T) {
+	err := setupEnv()
+	if err != nil {
+		t.Errorf("Test suite setup failed: %s", err.Error())
+	}
+
+	for _, tc := range podAddedTcs {
+		t.Run(tc.podName, func(t *testing.T) {
+			testSethandler := setupTestSethandler(tc.poolConf)
+			testSethandler.PodAdded(getPod(tc.podName))
+			cpusActual, err := readFakeCpusetFile(getPod(tc.podName))
+			if err != nil && !tc.isErrorExpectedAtFakeFileRead {
+				t.Logf("Could not process FAKE cpuset file because: %s", err.Error())
+			}
+			if !reflect.DeepEqual(cpusActual, tc.expectedCpus) {
+				t.Errorf("Mismatch in expected (%s) vs actual (%s) cpus written in cpuset file: ", tc.expectedCpus, cpusActual)
+			}
+		})
+	}
+	err = utils.RemoveTempSysFs()
+	if err != nil {
+		t.Logf("Removal of temp fs for cpusets was unsuccessful due to: %s", err.Error())
+	}
+}
+
+func TestPodChanged(t *testing.T) {
+	err := setupEnv()
+	if err != nil {
+		t.Errorf("Test suite could not be set up: %s", err.Error())
+	}
+
+	for _, tc := range podChangedTcs {
+		t.Run(tc.newPodName, func(t *testing.T) {
+			testSethandler := setupTestSethandler(tc.poolConf)
+			testSethandler.PodChanged(getPod(tc.oldPodName), getPod(tc.newPodName))
+			cpusActual, err := readFakeCpusetFile(getPod(tc.newPodName))
+			if err != nil && !tc.isErrorExpectedAtFakeFileRead {
+				t.Logf("Could not process FAKE cpuset file because: %s", err.Error())
+			}
+			if !reflect.DeepEqual(cpusActual, tc.expectedCpus) {
+				t.Errorf("Mismatch in expected (%s) vs actual (%s) cpus written in cpuset file:", tc.expectedCpus, cpusActual)
+			}
+
+		})
+	}
+	err = utils.RemoveTempSysFs()
+	if err != nil {
+		t.Logf("Removal of temp fs for cpusets was unsuccessful due to: %s", err.Error())
+	}
+}
+
+func setupEnv() error {
+	os.Setenv("NODE_NAME", "caas_master")
+	err := utils.CreateTempSysFs()
+	if err != nil {
+		return err
+	}
+	err = utils.CreateCheckpointFile()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func setupTestSethandler(poolConf types.PoolConfig) sethandler.SetHandler {
+	sh := sethandler.SetHandler{}
+	sh.SetSetHandler(poolConf, getFakeCpusetRoot(wildcardFakeCpuSetPath), fake.NewSimpleClientset())
+	return sh.SetHandler()
```

Comment:
`return sh` would work here ? -> `SetHandler()` is not needed.

Reply:
Indeed, it would work. I deliberately used `SetHandler()` to increase the coverage a bit. 
Nevertheless I will change it, if you think it would be better.

Reply:
I think it is better that way and then we are ready to merge it.

Reply:
Ok. Fixed.